### PR TITLE
chore: site test opt

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  -  |
| 🇨🇳 Chinese |  -  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 794af63</samp>

Improved check-site script by using `cheerio.load` instead of `cheerio` to parse HTML.

### <samp>🤖 Generated by Copilot at 794af63</samp>

* Import and use `cheerio.load` function to parse HTML strings and avoid memory leaks and performance issues ([link](https://github.com/ant-design/ant-design/pull/45351/files?diff=unified&w=0#diff-8f7f57c603d8bbb4d8c032eaceecbbd943fcc2c29eaea1d7b3c55e88b4ca1a3aL6-R6), [link](https://github.com/ant-design/ant-design/pull/45351/files?diff=unified&w=0#diff-8f7f57c603d8bbb4d8c032eaceecbbd943fcc2c29eaea1d7b3c55e88b4ca1a3aL24-R24))

### CODE REVIEW

File: scripts/check-site.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ describe('site test', () => {
   const render = async (path: string) => {
     const resp = await fetch(`http://127.0.0.1:${port}${path}`).then(async (res) => {
       const html = await res.text();
-      const $ = cheerio.load(html, { decodeEntities: false, recognizeSelfClosing: true });
+      const $ = load(html, { decodeEntities: false, recognizeSelfClosing: true });
```
Comment (li-jia-nan) [inline scripts/check-site.ts:24]: 为啥要改这个？

Comment (kiner-tang) [reply scripts/check-site.ts:24]: image

File: scripts/check-site.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ describe('site test', () => {
   const render = async (path: string) => {
     const resp = await fetch(`http://127.0.0.1:${port}${path}`).then(async (res) => {
       const html = await res.text();
```
Comment (li-jia-nan) [inline scripts/check-site.ts:23]:
```suggestion
      const html: string = await res.text();
```

Comment (kiner-tang) [reply scripts/check-site.ts:23]: 简单类型感觉不一定要加类型

Comment (li-jia-nan) [reply scripts/check-site.ts:23]: 能自动推导的就不需要加，这里推导出来是 any

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**